src/main/java/org/folio/circulation/resources/DueDateNotRealTimeScheduledNoticeProcessingResource.java

```diff
@@ -47,12 +47,15 @@ public DueDateNotRealTimeScheduledNoticeProcessingResource(HttpClient client) {
 
   @Override
   protected CompletableFuture<Result<MultipleRecords<ScheduledNotice>>> findNoticesToSend(
+    ConfigurationRepository configurationRepository,
     ScheduledNoticesRepository scheduledNoticesRepository, PageLimit pageLimit) {
 
-    DateTime timeLimit = LocalDate.now().toDateTime(LocalTime.MIDNIGHT);
-    return scheduledNoticesRepository.findNotices(timeLimit, false,
-      Collections.singletonList(TriggeringEvent.DUE_DATE),
-      FETCH_NOTICES_SORT_CLAUSE, pageLimit);
+    return configurationRepository.findTimeZoneConfiguration()
+      .thenApply(r -> r.map(tz -> ClockManager.getClockManager().getDateTime()
+        .withZone(tz).withTimeAtStartOfDay()))
```

Comment:
Is this converting the timezone into a date-time for the start of today in that timezone?

```suggestion
      .thenApply(r -> r.map(tz -> startOfTodayInTimezone(tz)))
```

Reply:
Exactly, start of today in tenant's timezone instead of UTC. Do you propose to extract a method here?

Reply:
> Do you propose to extract a method here?

Yeah, to raise the abstraction level of that statement and make the intent explicit

(the same for the other comment too)

Reply:
Done